source/slang/slang-emit.cpp

```diff
@@ -1460,6 +1513,13 @@ Result linkAndOptimizeIR(
     if (!targetProgram->getOptionSet().shouldPerformMinimumOptimizations())
         checkUnsupportedInst(codeGenContext->getTargetReq(), irModule, sink);
 
+    if (targetProgram->getOptionSet().getBoolOption(CompilerOptionName::EmbedDXIL))
+    {
+        // We need to remove all the functions that can't be represented
+        // in a DXIL library, eg. with resources.
+        SLANG_RETURN_ON_FAIL(removeNonEmbeddableDXIL(irModule));
```

Comment:
I don't think you can just remove these functions, because I can have a module that is:
```
internal void nonEmbedable(Texture2D t) { ... }

public void embeddable() { nonEmbeddable(...); }
```

All we can do is to say we don't mark `nonEmbeddable` as `export` when emitting HLSL.